Userland/Libraries/LibGUI/Statusbar.cpp

```diff
@@ -38,6 +38,19 @@ NonnullRefPtr<Statusbar::Segment> Statusbar::create_segment()
     return widget;
 }
 
+void Statusbar::child_event(Core::ChildEvent& event)
+{
+    auto& event_to_forward = event;
+    // To ensure that the ResizeCorner is always the last widget, and thus stays in the corner,
+    // we replace ChildAdded events that do not request specific placement with events that request placement before the corner
+    if (event.type() == Event::ChildAdded && is<Widget>(*event.child()) && !event.insertion_before_child()) {
+        Core::ChildEvent new_event(Event::ChildAdded, *event.child(), m_corner.ptr());
+        event_to_forward = new_event;
+    }
+
+    return Widget::child_event(event_to_forward);
```

Comment:
Slightly confusing doing `return Widget::child_event` here when `child_event()` returns void, and this is not an early return. 
```suggestion
    Widget::child_event(event_to_forward);
```

Reply:
That's the way it is done throughout the system with overloaded event handlers. Just look at `child_event()` in `Widget` and `Core::Object`.